manager/src/main/java/org/openremote/manager/security/KeyStoreServiceImpl.java

```diff
@@ -55,66 +71,83 @@ public void init(Container container) throws Exception {
+
+		String keyStorePassword = getString(container.getConfig(), OR_SSL_CLIENT_KEYSTORE_PASSWORD, String.valueOf(getKeyStorePassword()));
+		String trustStorePassword = getString(container.getConfig(), OR_SSL_CLIENT_KEYSTORE_PASSWORD, String.valueOf(getKeyStorePassword()));
 
-			getLogger().fine("Accessing: " + storePath);
-			KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
-			File keyStoreFile = storePath.toAbsolutePath().toFile();
 
-			// Ensure parent directories exist
-			if (keyStoreFile.getParentFile() != null && !keyStoreFile.getParentFile().exists()) {
-				keyStoreFile.getParentFile().mkdirs();
+		if(keyStorePath.isPresent()){
+			try{
+				keyStore = KeyStore.getInstance(keyStorePath.get().toFile(), keyStorePassword.toCharArray());
+			} catch (Exception e){
+				throw e;
 			}
+		}else{
+			Path defaultKeyStorePath = persistenceService.resolvePath(Paths.get("keystores").resolve("client_keystore.p12"));
+			if(new File(defaultKeyStorePath.toUri()).exists()) {
+				keyStorePath = Optional.of(defaultKeyStorePath);
+				this.keyStore = KeyStore.getInstance(new File(defaultKeyStorePath.toUri()), getKeyStorePassword());
+			}else{
+				keyStore = createKeyStore(defaultKeyStorePath);
+			}
+
+			this.keyStorePath = defaultKeyStorePath;
+		}
 
-			// Initialize KeyStore
-			try (FileInputStream fis = new FileInputStream(keyStoreFile)) {
-				keyStore.load(fis, getKeyStorePassword()); // Assuming keystorePassword is defined elsewhere
-			} catch (Exception e) {
-				getLogger().severe("Couldn't find KeyStore file " + storePath + ", initializing new KeyStore file");
-				keyStore.load(null, getKeyStorePassword());
-				// Save the newly created KeyStore
-				try (OutputStream os = new FileOutputStream(keyStoreFile)) {
-					keyStore.store(os, getKeyStorePassword());
-				} catch (Exception saveException) {
-					getLogger().severe("Couldn't store KeyStore to Storage! " + saveException.getMessage());
-				}
+		if(trustStorePath.isPresent()){
+			try{
+				keyStore = KeyStore.getInstance(trustStorePath.get().toFile(), trustStorePassword.toCharArray());
+			} catch (Exception e){
+				throw e;
+			}
+		}else{
+			Path defaultTrustStorePath = persistenceService.resolvePath(Paths.get("keystores").resolve("client_truststore.p12"));
+			if(new File(defaultTrustStorePath.toUri()).exists()) {
+				trustStorePath = Optional.of(defaultTrustStorePath);
+				this.trustStore = KeyStore.getInstance(new File(defaultTrustStorePath.toUri()), getKeyStorePassword());
+			}else{
+				trustStore = createKeyStore(defaultTrustStorePath);
 			}
+			this.trustStorePath = defaultTrustStorePath;
+		}
+	}
+	@Override
+	public KeyStore getKeyStore() {
+		try {
+			return KeyStore.getInstance(this.keyStorePath.toFile(), getKeyStorePassword());
+		} catch (Exception e) {
+			throw new RuntimeException(e);
 		}
 	}
 	@Override
-	public KeyStore getKeyStore(KeyStoreType type) {
-		Path storePath = type.getPath();
+	public KeyStore getTrustStore() {
```

Comment:
That's because there's no actual merging of keystores happening, because the SPI I had to create just considers all keystores supplied to it, in `org.openremote.manager.security.KeyStoreServiceImpl#getTrustManagerFactory`.